CMakeLists.txt

```diff
@@ -204,7 +213,7 @@ find_package(Threads REQUIRED)
 find_package(ZLIB REQUIRED)
 
 if (TON_ARCH AND NOT MSVC)
-  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -march=${TON_ARCH}")
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -mcpu=${TON_ARCH}")
```

Comment:
We should keep `-march` flag for `x86` arch. Source: https://stackoverflow.com/a/42722790/4253144